server/calendar.ts

```diff
@@ -0,0 +1,79 @@
+import 'moment-timezone';
+
+import express from 'express';
+import ical from 'ical-generator';
+import moment from 'moment';
+
+import eagerIterator from './eager-iterator';
+import { getSportIcon } from './sport-icons';
+import { Strava } from './strava';
+
+export default function calendarRouter(domain: string): express.Router {
+  const router = express.Router();
+
+  const mkRequestLogin = (res: express.Response) => async () => {
+    res.status(403).send('Not logged in');
+  };
+
+  router.get('/', (req: express.Request, res: express.Response) => {
+    const token = req.cookies.token;
+    const requestLogin = mkRequestLogin(res);
+    if (token && new Strava(domain, token, requestLogin).hasToken) {
+      const path = req.get('Host') + req.originalUrl.replace(/\/?$/, '/');
+      const fullPath = `webcal://${path}${req.cookies.token}.ics`;
+      res.redirect(fullPath);
+    } else {
+      requestLogin();
+    }
+  });
+
+  router.get('/:token.ics', async (req: express.Request, res: express.Response) => {
+    const strava = new Strava(domain, req.params.token, mkRequestLogin(res));
+
+    res.type('text/calendar; charset=UTF-8');
+
+    const cal = ical({ name: 'Strava activities' });
```

Comment:
- [ ] Add proper caching of ical (only fetch new activities, e.g. from past week)
- [ ] Clear old caches after certain time
- [ ] Handle logout through GUI
- [ ] Add link to GUI